### ISSUE TITLE
[Bug]: Dataview cannot be rendered in Kanban

### ISSUE THREAD

**Issue by Missionary2112** [mgmeyers/obsidian-kanban#550]
### Describe the bug

I use Dataview to display a list for files. I embedded a file with Dataview List in Kanban card and it would appear the list there. Just recently, the list appears blank.

### Expected Behavior

It should display embedded file with Dataview List

### Steps to reproduce

Step 1: Create a new file with Dataview List
Step 2: Embed the file in Kanban card
Step 3: See error: The list is blank

### Operating system

macOS

**Comment by AB1908** [comment]
Duplicate of #345, #467, #85, #148, #321.

**Comment by Yunike08** [comment]
Oh yes, please address this soon. Kanban and Dataview working together is amazing!

**Bug**
In my case, I also use Dataview Tables inside my Kanban cards. However, ever since I updated Dataview from 0.4.26 to 0.5.36, all my Dataview content inside the cards are empty. It does mention how many items are inside the table but not the actual content (birthdays, events, story characters, etc.)

The code I use for this example is

```dataview
Table without ID
link(file.link, Aliases) AS "Name", Description
From #TC/Plot/CList/Character
Where Type = "storyChar"
Sort file.name
```

![03](https://user-images.githubusercontent.com/106167889/175777075-d0bb4c83-175d-4c4e-a10a-319602ab0505.PNG)
.

If I, however, revert Dataview back to 0.4.26, I get back my content.

**Expected Behavior**
(with the additional number of how many items are there in the table)
Dataview table can work inside Kanban board.
![IMG_20220625_221223](https://user-images.githubusercontent.com/106167889/175777353-f9bea7d2-c524-43a6-8115-cc2d7fe964b2.jpg)

That's the status I'm currently in now. I hope they will work together again.

**System**
Mobile user Android v12

**Comment by AB1908** [comment]
I have a hunch on what's happening here but not the technical expertise to pin down what/why. I hope mgmeyers can find the time to look at this. Dataview recently changed its rendering to be React based FWIW.

**Comment by kometenstaub** [comment]
> I have a hunch on what's happening here but not the technical expertise to pin down what/why. I hope mgmeyers can find the time to look at this. Dataview recently changed its rendering to be React based FWIW.

What is your hunch?

**Comment by kometenstaub** [comment]
@AB1908 (forgot to tag you above)

**Comment by AB1908** [comment]
I assumed it was related to the order of rendering. Dataview renders before the markdown post processor, at least IIRC, which is what may be interfering here. But if this always happened, the embeds should never have worked in the first place.

TL;DR: I have no clue

**Comment by salievi** [comment]
Bizarrely, it is working some of the time. If I put the Dataview code in a card and hit return, the code doesn't render (usually it just says "Loading" forever. But if I enter the field to edit it again, then close it using the X on the card (not Enter), it renders. Does this imply a solution?

**Comment by birgittealstrom** [comment]
> Bizarrely, it is working some of the time. If I put the Dataview code in a card and hit return, the code doesn't render (usually it just says "Loading" forever. But if I enter the field to edit it again, then close it using the X on the card (not Enter), it renders. Does this imply a solution?

I experience exactly the same.

**Comment by joshwayne** [comment]
I'll kick in a $100 bounty for this feature

**Comment by shafferchance** [comment]
So it seems part of the reason it will work sometimes and not others comes down to if the query is ready when the List formatter runs on the View for this file. Going to see about making a branch on my fork that waits for these to resolves as a first step towards #345. Which in turn would resolve this issue and act as a way to atleast let our swim lanes be populated via metadata. Ideally transitions would also allow for metadata updates in the future but that is out of scope of this issue.

**Comment by shafferchance** [comment]
Alright, I'm back again, after some more digging found that the data is coming out of `renderMarkdown` as expected leaving the issue to React State as both technically use React (however this is more for dataviewjs). So we need a way to connect the two instances. Probably a lift upon a new child being added to the KanbanView by Dataview. If we're working with DQL the nice piece is that we can parse it after and push it into the dom returned therefore eliminating the two separate state issues.

**Comment by shafferchance** [comment]
Upon disabling `Mark` from inside of the `MarkdownRenderer` component the dataviewjs query renders correctly.

![image](https://github.com/mgmeyers/obsidian-kanban/assets/3750436/1bb02826-9aa5-493c-b08e-9e9f0297ba80)
Will look into making these work together, hunch here is the `cloneNode`method is not quite working deeply and/or it disconnects the DataviewJsRenderer from its DOM nodes.

**Comment by shafferchance** [comment]
Re-enabled mark but removed the cloneNode(true) and now things are working as expected.

@mgmeyers Why was a deepClone used here? Would either removing it entirely or creating a handler of a case with cards containing dataviews aren't clone be an acceptable solution?

**Comment by shafferchance** [comment]
Ah while testing I think I found the purpose of the clone, looks like if I drag and drop a list all the cards content disappears until I refresh.

**Comment by Kribbage** [comment]
Is the setting for putting tags at the bottom of the card enabled on this board? I noticed that it yanks and removes any tags in my dataview tables if I have that enabled, was going to submit a report about that, unless this is the same thing.

**Comment by shafferchance** [comment]
I don't it enabled but we can add it to the scope here for sure

2024年5月22日(水) 2:44 Kribby ***@***.***>:
